adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java

```diff
@@ -1148,14 +1149,92 @@ protected final void onCommandReceived(
         final String commandRequestId = (command.isOneWay() ? "" : command.getRequestId());
         final String topic = String.format("%s/%s/%s/%s/%s/%s", subscription.getEndpoint(), tenantId, deviceId,
                 subscription.getRequestPart(), commandRequestId, command.getName());
-        endpoint.publish(topic, command.getPayload(), qos, false, false);
+        LOG.debug("Publishing command to device [tenant-id: {}, device-id: {}, MQTT client-id: {}, QoS: {}]",
+                        subscription.getTenant(), subscription.getDeviceId(), subscription.getClientId(), subscription.getQos());
+        commandContext.getCurrentSpan().log(String.format("Publishing command to device [tenant-id: %s, device-id: %s, MQTT client-id: %s, QoS: %s]",
```

Comment:
the tenant and device ID will already be present in the span as tags. Please use a Map to log the event, client ID and QoS in a structured way

Reply:
Done